src/codegen/generate.ts

```diff
@@ -706,36 +681,46 @@ export default class ApiGenerator {
     return type;
   }
 
-  /**
-   * Check if schema is readOnly or has readOnly properties
-   */
-  isSchemaReadOnly(schema: SchemaObject | OpenAPIV3.ReferenceObject) {
+  checkSchemaOnlyMode(
```

Comment:
merged `isSchemaReadOnly` and `isSchemaWriteOnly` and will now consider a schema to have write/read only qualities when any deeply nested prop has the feature